packages/braid-design-system/src/lib/components/TextLink/TextLink.tsx

```diff
@@ -102,9 +103,38 @@ export const useLinkStyles = ({
   );
 };
 
+export const TextLinkContent = ({
+  icon,
+  iconPosition = 'leading',
+  children,
+}: Pick<TextLinkProps, 'icon' | 'iconPosition' | 'children'>) => (
+  <>
```

Comment:
Do we want make sure the icon doesn't break on a new line?

<img width="339" alt="wrap" src="https://user-images.githubusercontent.com/3297808/223601184-de4820cb-993b-49d5-a161-a10ff9c9c9a8.png">